django/db/models/query_utils.py

```diff
@@ -537,13 +537,41 @@ def relabeled_clone(self, change_map):
 
     def resolve_expression(self, query, reuse, *args, **kwargs):
         clone = self.clone()
-        clone.resolved_condition = query.build_filter(
-            self.condition,
-            can_reuse=reuse,
```

Comment:
Shouldn't this be a matter of augmenting the `can_reuse` with aliases that should be reused instead?